From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ROCKNIX <rocknix@rocknix.org>
Date: Sat, 15 Mar 2026 12:00:00 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3562 remove opp-supported-hw from GPU OPP table

Remove opp-supported-hw property from GPU OPP table entries.
This property requires nvmem cells to determine hardware version,
and if nvmem read fails or returns unexpected values, all OPPs are
filtered out causing "no supported OPPs" error.

Signed-off-by: ROCKNIX <rocknix@rocknix.org>
---
 arch/arm64/boot/dts/rockchip/rk3562.dtsi | 7 -------
 1 file changed, 7 deletions(-)

--- a/arch/arm64/boot/dts/rockchip/rk3562.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3562.dtsi
@@ -1806,27 +1806,22 @@
 		rockchip,low-temp-min-volt = <925000>;
 
 		opp-300000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <300000000>;
 			opp-microvolt = <825000 825000 1000000>;
 		};
 		opp-400000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <400000000>;
 			opp-microvolt = <825000 825000 1000000>;
 		};
 		opp-500000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <500000000>;
 			opp-microvolt = <825000 825000 1000000>;
 		};
 		opp-600000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <825000 825000 1000000>;
 		};
 		opp-700000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <700000000>;
 			opp-microvolt = <900000 900000 1000000>;
 			opp-microvolt-L0 = <900000 900000 1000000>;
@@ -1836,7 +1831,6 @@
 			opp-microvolt-L4 = <825000 825000 1000000>;
 		};
 		opp-800000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <950000 950000 1000000>;
 			opp-microvolt-L0 = <950000 950000 1000000>;
@@ -1846,7 +1840,6 @@
 			opp-microvolt-L4 = <850000 850000 1000000>;
 		};
 		opp-900000000 {
-			opp-supported-hw = <0xf9 0xffff>;
 			opp-hz = /bits/ 64 <900000000>;
 			opp-microvolt = <1000000 1000000 1000000>;
 			opp-microvolt-L0 = <1000000 1000000 1000000>;
